File: PythonSolutions/CountAndReplaceLetters.py
def duplicate_encode(word):
    d = dict()
    for c in word.lower():
        if c in d:
            d[c] += 1
        else:
            d[c] = 1
    
    result = ""
    for c in word.lower():
        if d[c] > 1:
            result += ")"
        else:
            result += "("
    
    return result

File: CountAndReplaceLetters.cs
using System.Collections.Generic;
public class CountAndReplaceLetters
{
  public static string DuplicateEncode(string word)
  {
    Dictionary<char, int> d = new Dictionary<char, int>();
    
    foreach(var c in word.ToLower()) {
      if (d.ContainsKey(c)) {
        d[c]++;
      }
      else {
        d.Add(c, 1);
      }
    }
    string result = "";
    foreach(var x in word.ToLower()) {
        result += (d[x] > 1) ? ")" : "(";
    }
    return result;
  }
}